debezium-core/src/main/java/io/debezium/function/BufferedBlockingConsumer.java

```diff
@@ -42,27 +43,31 @@ public default void flush() throws InterruptedException {
      * and buffer the latest.
      * <p>
      * The resulting consumer is not threadsafe.
-     * 
+     *
      * @param delegate the delegate to which values should be flushed; may not be null
      * @return the blocking consumer that buffers a single value at a time; never null
      */
     public static <T> BufferedBlockingConsumer<T> bufferLast(BlockingConsumer<T> delegate) {
         return new BufferedBlockingConsumer<T>() {
-            private T last;
+
+            private final AtomicReference<T> last = new AtomicReference<>();
 
             @Override
             public void accept(T t) throws InterruptedException {
-                if (last != null) delegate.accept(last);
-                last = t;
+                T previous = last.getAndSet(t);
+                if (previous != null) {
+                    delegate.accept(previous);
+                }
             }
 
             @Override
-            public void flush(Function<T, T> function) throws InterruptedException {
-                if (last != null) {
+            public synchronized void flush(Function<T, T> function) throws InterruptedException {
```

Comment:
Is the method `synchronized intentionally`?

Reply:
Yes, I thought it'd be good to prevent concurrent `flush()` calls.